server/src/utils/getConfig.ts

```diff
@@ -12,36 +12,84 @@ import {
     communicationServicesConnectionString:
-      process.env[VV_COMMUNICATION_SERVICES_CONNECTION_STRING] ?? DefaultConfig.communicationServicesConnectionString,
-    microsoftBookingsUrl: process.env[VV_MICROSOFT_BOOKINGS_URL_ENV_NAME] ?? DefaultConfig.microsoftBookingsUrl,
+      process.env[VV_COMMUNICATION_SERVICES_CONNECTION_STRING] ?? defaultConfig.communicationServicesConnectionString,
+    microsoftBookingsUrl: process.env[VV_MICROSOFT_BOOKINGS_URL_ENV_NAME] ?? defaultConfig.microsoftBookingsUrl,
     chatEnabled:
       typeof process.env[VV_CHAT_ENABLED_ENV_NAME] === 'string'
         ? process.env[VV_CHAT_ENABLED_ENV_NAME]?.toLowerCase() === 'true'
-        : DefaultConfig.chatEnabled,
+        : defaultConfig.chatEnabled,
     screenShareEnabled:
       typeof process.env[VV_SCREENSHARE_ENABLED_ENV_NAME] === 'string'
         ? process.env[VV_SCREENSHARE_ENABLED_ENV_NAME]?.toLowerCase() === 'true'
-        : DefaultConfig.screenShareEnabled,
-    companyName: process.env[VV_COMPANY_NAME_ENV_NAME] ?? DefaultConfig.companyName,
-    colorPalette: process.env[VV_COLOR_PALETTE_ENV_NAME] ?? DefaultConfig.colorPalette,
-    waitingTitle: process.env[VV_WAITING_TITLE_ENV_NAME] ?? DefaultConfig.waitingTitle,
-    waitingSubtitle: process.env[VV_WAITING_SUBTITLE_ENV_NAME] ?? DefaultConfig.waitingSubtitle,
-    logoUrl: process.env[VV_LOGO_URL_ENV_NAME] ?? DefaultConfig.logoUrl
-  };
+        : defaultConfig.screenShareEnabled,
+    companyName: process.env[VV_COMPANY_NAME_ENV_NAME] ?? defaultConfig.companyName,
+    colorPalette: process.env[VV_COLOR_PALETTE_ENV_NAME] ?? defaultConfig.colorPalette,
+    waitingTitle: process.env[VV_WAITING_TITLE_ENV_NAME] ?? defaultConfig.waitingTitle,
+    waitingSubtitle: process.env[VV_WAITING_SUBTITLE_ENV_NAME] ?? defaultConfig.waitingSubtitle,
+    logoUrl: process.env[VV_LOGO_URL_ENV_NAME] ?? defaultConfig.logoUrl
+  } as ServerConfigModel;
+
+  config.postCall = getPostCallConfig(defaultConfig);
+
+  return config;
+};
+
+const isValidPostCallSurveyType = (postcallSurveyType: string): postcallSurveyType is PostCallSurveyType => {
+  return ['msforms', 'thirdparty'].indexOf(postcallSurveyType) !== -1;
+};
+
+const getPostCallConfig = (defaultConfig: ServerConfigModel): PostCallConfig | undefined => {
+  let postcallConfig: PostCallConfig | undefined;
+  //Setting values for postcallconfig from defaultconfig values first (if valid)
+  if (
+    defaultConfig.postCall?.survey?.type &&
+    isValidPostCallSurveyType(defaultConfig.postCall?.survey?.type) &&
+    defaultConfig.postCall?.survey?.options?.surveyUrl
+  ) {
+    postcallConfig = {
+      survey: {
+        type: defaultConfig.postCall?.survey?.type,
+        options: { surveyUrl: defaultConfig.postCall?.survey?.options?.surveyUrl }
+      }
+    };
+  }
+
+  try {
+    const postcallSurveyType = process.env[VV_POSTCALL_SURVEY_TYPE_ENV_NAME];
+    const postcallSurveyUrl = process.env[VV_POSTCALL_SURVEY_OPTIONS_SURVEYURL_ENV_NAME];
```

Comment:
I think this try/catch is a great starting point. The code from lines 57-68 very closely resembles the code in this try catch. Can we combine them like so?

```
const postCallSurveyType = process.env[VV_POSTCALL_SURVEY_TYPE_ENV_NAME] ?? defaultConfig.postCall?.survey?.type;
const postCallSurveyUrl = process.env[VV_POSTCALL_SURVEY_OPTIONS_SURVEYURL_ENV_NAME] ?? defaultConfig.postCall?.survey?.options?.surveyUrl;

...
rest of validation logic
...
```

This way we just have one single code block that does the validation instead of two.